packages/@lwc/engine-core/src/framework/modules/static-parts.ts

```diff
@@ -128,5 +141,29 @@ export function patchStaticParts(n1: VStatic, n2: VStatic) {
         part.elm = prevParts![i].elm;
         // Refs must be updated after every render due to refVNodes getting reset before every render
         applyRefs(part, currPartsOwner);
+        patchAttributes(prevParts![i], part, renderer);
+    }
+}
+
+/**
+ * Mounts the hydration specific attributes
+ * @param vnode the parent VStatic node
+ * @param renderer the renderer to use
+ */
+export function hydrateStaticParts(vnode: VStatic, renderer: RendererAPI): void {
+    if (!process.env.IS_BROWSER) {
+        return;
+    }
+
+    const { parts, owner } = vnode;
+    if (isUndefined(parts)) {
+        return;
+    }
+
+    for (const part of parts) {
+        // Event listeners only need to be applied once when mounting
+        applyEventListeners(part, renderer);
+        // Refs must be updated after every render due to refVNodes getting reset before every render
+        applyRefs(part, owner);
```

Comment:
Why don't we handle attributes here? Because, if there is no hydration mismatch, then we assume that SSR gave us the correct attributes?

Reply:
Yup, exactly. If there are no hydration mismatches then there shouldn't be a need to patch attributes (because they will be the same).

I'll add a comment explaining it here.